src/actions/attack.cpp

```diff
@@ -157,21 +157,27 @@ battle_context_unit_stats::battle_context_unit_stats(const unit& u,
 	}
 
 	// Compute chance to hit.
-	chance_to_hit = opp.defense_modifier(resources::gameboard->map().get_terrain(opp_loc)) + weapon->accuracy()
+    int tmp_chance_to_hit = opp.defense_modifier(resources::gameboard->map().get_terrain(opp_loc)) + weapon->accuracy()
 		- (opp_weapon ? opp_weapon->parry() : 0);
 
-	if(chance_to_hit > 100) {
-		chance_to_hit = 100;
+	if(tmp_chance_to_hit > 100) {
+		tmp_chance_to_hit = 100;
 	}
```

Comment:
Here we should check for negative values too, in case there's parry. You can use `utils::clamp` for this.